### PR TITLE
[PSyIR-FE] allow empty program

### PR DESCRIPTION
**Pull request by arporter**
Completes #1723 by adding testing and authorship info.

### CODE REVIEW

**Comment by arporter** [comment]
@schreiberx I'm struggling to reproduce an "empty program". Could you give me an example of the Fortran that caused you trouble?

**Comment by schreiberx** [comment]
[t.zip](https://github.com/stfc/PSyclone/files/8747205/t.zip)
Please find a reproducer including the full error in this zip file.

File: src/psyclone/psyir/frontend/fparser2.py [hunk 1]
```diff
@@ -30,16 +30,16 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 # -----------------------------------------------------------------------------
-# Authors R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
-# Modified I. Kavcic, Met Office
-# Author: J. Henrichs, Bureau of Meteorology
+# Authors: R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
+#          J. Henrichs, Bureau of Meteorology
+#          I. Kavcic, Met Office
+#          M. Schrieber, Université Grenoble Alpes
```
Comment (TeranIvy) [inline src/psyclone/psyir/frontend/fparser2.py:36]: I think Martin's last name is "Schreiber".

**Comment by arporter** [comment]
Aha, thanks very much @schreiberx. Ok, now I see that `psyir_from_file` is not working for an empty file. It should return an empty FileContainer node. Also, I'll try to spell your name correctly ;-) (Thanks Iva.)

**Comment by arporter** [comment]
This is ready for a first look now. It boils down to two small bug fixes in the frontend to allow for the case where the parse tree is empty. One for @sergisiso  or @hiker I think.

**Review by sergisiso — APPROVED** [review]
All new lines are covered by untit tests, and all test and examples pass. There is no pylcodestyle or pylint warnings introduced. This is a bugfix so not new documentation is needed.

There are a couple of conflicts to merge with master, but I will fix those myself. PR approved for merging.

**Pull request merged**